File: app/src/main/java/com/notai/data/repository/TagRepository.java
package com.notai.data.repository;

import androidx.lifecycle.LiveData;

import com.notai.data.dao.TagDao;
import com.notai.data.model.Tag;

import java.util.List;
import java.util.concurrent.Executor;
import java.util.concurrent.Executors;

/**
 * 标签仓库类（数据层）
 */
public class TagRepository {
    private final TagDao tagDao;
    private final Executor executor;

    public TagRepository(TagDao tagDao) {
        this.tagDao = tagDao;
        this.executor = Executors.newSingleThreadExecutor();
    }

    public LiveData<List<Tag>> getAllTags() {
        return tagDao.getAllTags();
    }

    public Tag getTagById(long tagId) {
        return tagDao.getTagById(tagId);
    }

    public Tag getTagByName(String name) {
        return tagDao.getTagByName(name);
    }

    public void insertTag(Tag tag) {
        executor.execute(() -> tagDao.insertTag(tag));
    }

    public void updateTag(Tag tag) {
        executor.execute(() -> tagDao.updateTag(tag));
    }

    public void deleteTag(long tagId) {
        executor.execute(() -> {
            // 先删除关联关系
            tagDao.deleteNoteTagRefsByTagId(tagId);
            // 再删除标签
            tagDao.deleteTagById(tagId);
        });
    }
}

File: app/src/main/java/com/notai/data/repository/TagRepository.kt
package com.notai.data.repository

import com.notai.data.dao.TagDao
import com.notai.data.model.Tag
import kotlinx.coroutines.flow.Flow

/**
 * 标签仓库类（数据层）
 */
class TagRepository(
    private val tagDao: TagDao
) {
    
    fun getAllTags(): Flow<List<Tag>> = tagDao.getAllTags()
    
    suspend fun getTagById(tagId: Long): Tag? = tagDao.getTagById(tagId)
    
    suspend fun getTagByName(name: String): Tag? = tagDao.getTagByName(name)
    
    suspend fun insertTag(tag: Tag): Long = tagDao.insertTag(tag)
    
    suspend fun updateTag(tag: Tag) = tagDao.updateTag(tag)
    
    suspend fun deleteTag(tagId: Long) {
        // 先删除关联关系
        tagDao.deleteNoteTagRefsByTagId(tagId)
        // 再删除标签
        tagDao.deleteTagById(tagId)
    }
}

